src/plugins/intel_gpu/src/kernel_selector/kernels/gemm/gemm_kernel_tiled_opt.cpp

```diff
@@ -328,6 +328,11 @@ JitConstants GemmKernelTiledOpt::GetJitConstants(const gemm_params& params) cons
                 if ((vec_axis_dim == 1 && op.tensors[0].LogicalSize() != 1) && (params.inputs[1].X().v != vec_axis_dim)) {
                     vec_load_type = LoadType::LT_UNALIGNED;
                 }
+
+                // For innermost dimension, switch to load through scalar method
```

Comment:
Let's make the comment more verbose, like:
`In case of a fused operation where input data has dynamic innermost dimension, we shouldn't use block reads, as the dynamic dimension may only have a single broadcastable value. Therefore, we should force scalar fusions (to read elements one by one using _SAFE index calculation) and apply them in the loop over vec_size`